include/unifex/unstoppable.hpp

```diff
@@ -0,0 +1,213 @@
+  using operation = unstoppable_operation<Source, Receiver>;
+public:
+  explicit unstoppable_source_receiver(operation* op) noexcept
+  : op_(op) {}
+
+  unstoppable_source_receiver(unstoppable_source_receiver&& other) noexcept
+  : op_(std::exchange(other.op_, {}))
+  {}
+ 
+  void set_value() {
+    assert(op_ != nullptr);
+    unifex::set_value(std::move(op_->receiver_));
+  }
+
+  template<
+    typename R = Receiver,
+    std::enable_if_t<std::is_invocable_v<decltype(unifex::set_done), R>, int> = 0>
+  void set_done() noexcept {
+    assert(op_ != nullptr);
+    unifex::set_value(std::move(op_->receiver_));
+  }
+
+  template<
+    typename Error,
+    std::enable_if_t<std::is_invocable_v<decltype(unifex::set_error), Receiver, Error>, int> = 0>
+  void set_error(Error&& error) noexcept {
+    assert(op_ != nullptr);
+    unifex::set_error(std::move(op_->receiver_), (Error&&)error);
+  }
+
+private:
+  template<typename CPO, typename... Args>
+  friend auto tag_invoke(CPO cpo, const unstoppable_source_receiver& r, Args&&... args)
+      noexcept(std::is_nothrow_invocable_v<CPO, const Receiver&, Args...>)
+      -> std::invoke_result_t<CPO, const Receiver&, Args...> {
+    return std::move(cpo)(r.get_receiver(), (Args&&)args...);
+  }
+  
+  template <typename VisitFunc>
+  friend void tag_invoke(
+      tag_t<visit_continuations>,
+      const unstoppable_source_receiver& r,
+      VisitFunc&& func) noexcept(std::is_nothrow_invocable_v<
+                                VisitFunc&,
+                                const Receiver&>) {
+    std::invoke(func, r.get_receiver());
+  }
+
+  const Receiver& get_receiver() const noexcept {
+    assert(op_ != nullptr);   
+    return op_->receiver_;
+  }
+
+  operation* op_;
+};
+
+template<typename Source, typename Receiver>
+class unstoppable_operation {
+  using source_receiver = detail::unstoppable_source_receiver<Source, Receiver>;
+
+public:
+  template<typename Source2, typename Receiver2>
+  explicit unstoppable_operation(Source2&& source, Receiver2&& receiver)
+      noexcept(std::is_nothrow_constructible_v<Source, Source2> &&
+               std::is_nothrow_constructible_v<Receiver, Receiver2> &&
+               is_nothrow_connectable_v<Source&, source_receiver>)
+  : source_((Source2&&)source)
+  , receiver_((Receiver&&)receiver)
+  {
+    sourceOp_.construct_from([&] {
+        return unifex::connect(source_, source_receiver{this});
+      });
+  }
+
+  ~unstoppable_operation() {
+    if (isSourceOpConstructed_) {
+      sourceOp_.destruct();
+      isSourceOpConstructed_ = false;
+    }
+  }
+
+  void start() & noexcept {
+    unifex::start(sourceOp_.get());
+  }
+
+private:
+  friend unstoppable_source_receiver<Source, Receiver>;
+
+  using source_op_t = operation_t<Source&, unstoppable_source_receiver<Source, Receiver>>;
+
+  UNIFEX_NO_UNIQUE_ADDRESS Source source_;
+  UNIFEX_NO_UNIQUE_ADDRESS Receiver receiver_;
+  bool isSourceOpConstructed_ = true;
```

Comment:
This flag seems unnecessary. The operation is never set to false other than in the destructor.
So the conditional destruction of sourceOp_ should not be necessary.

Also, you shouldn't need to store `source_` here either since it's only used in the call to `connect()` to construct the `sourceOp_`.

You can probably actually get away with not defining your own operation-state at all here, and instead just defining a receiver-wrapper type and forward that on to `connect()` on the source. This algorithm only translates receiver signals so it should be implementable entirely using a receiver that holds an inner receiver.